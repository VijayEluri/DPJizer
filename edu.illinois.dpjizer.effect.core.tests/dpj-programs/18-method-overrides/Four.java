/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class A<region R1> {
	A() {
	}

	void m() {

	}
}

class B extends A<Root> {
	region r;
	int a in r;
	int b in Root;

	B() {
	}

	void m() {
		a = 0;
		b = 1;
	}
}