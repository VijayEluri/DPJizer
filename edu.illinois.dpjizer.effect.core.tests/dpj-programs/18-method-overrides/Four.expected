A.A() invokes java.lang.Object.Object()
A.m() pure
B.B() invokes A.A() where { R1 -> Root }
B.m() writes Root, Root : B.r overrides A.m() where { R1 -> Root }
